src/xdocs/writingchecks.xml

```diff
@@ -679,14 +688,25 @@ public class MethodLimitCheck extends AbstractCheck
           <a href="https://github.com/checkstyle/checkstyle/issues/10629">
           Issue #10629.</a>
         </li>
-        <li>To get valid violations, code has to be compilable.
-          If it is not, you can get hard to understand parse errors.</li>
         <li>You cannot determine the type of an expression.
         Example: "getValue() + getValue2()"</li>
         <li>You cannot determine the full inheritance hierarchy of type.</li>
-        <li>You cannot see the contents of other files. You have access to the
+      </ul>
+      <p>
+        In addition to Java files, there are similar limitations that apply to all
+        type of text files:
```

Comment:
So, we can't check XML or properties file with a BOM (linelength, etc)?

Reply:
See the end of https://github.com/checkstyle/checkstyle/pull/11738#discussion_r897464531 and my example.

We can, but it requires manually weeding it out. The main point is, Checkstyle will not do this for you. If a 3rd party library like checkstyle extras doesn't take the extra effort to do this personally, then their grammars won't be able to parse those files either.

Reply:
Example on Windows:
````
package com.puppycrawl.tools.checkstyle;

import java.io.BufferedWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class CreateJavaBom {

    public static void main(String... args) throws IOException {
        try (BufferedWriter bw = Files.newBufferedWriter(Paths.get("M:\\JavaBOM.java"))) {
            bw.write("\ufeff"); // UTF8 BOM
            bw.write("public class TestClass {");
            bw.newLine();
            bw.write("    void method() {");
            bw.newLine();
            bw.write("    }");
            bw.newLine();
            bw.write("}");
            bw.newLine();
        }

        try (BufferedWriter bw = Files.newBufferedWriter(Paths.get("M:\\config.xml"))) {
            bw.write("<?xml version=\"1.0\"?>\r\n" //
                    + "<!DOCTYPE module PUBLIC\r\n" //
                    + "          \"-//Puppy Crawl//DTD Check Configuration 1.3//EN\"\r\n" //
                    + "          \"https://checkstyle.org/dtds/configuration_1_3.dtd\">\r\n" //
                    + "\r\n" //
                    + "<module name=\"Checker\">\r\n" //
                    + "    <property name=\"charset\" value=\"UTF-8\"/>\r\n" //
                    + "\r\n" //
                    + "    <module name=\"LineLength\">\r\n" //
                    + "    <property name=\"max\" value=\"0\"/>\r\n" //
                    + "    </module>\r\n" //
                    + "</module>");
        }

        Main.main("-c", "M:\\config.xml", "M:\\JavaBOM.java");
    }

}
````

Untouched Violations:
````
Starting audit...
[ERROR] M:\JavaBOM.java:1: Line is longer than 0 characters (found 25). [LineLength]
[ERROR] M:\JavaBOM.java:2: Line is longer than 0 characters (found 19). [LineLength]
[ERROR] M:\JavaBOM.java:3: Line is longer than 0 characters (found 5). [LineLength]
[ERROR] M:\JavaBOM.java:4: Line is longer than 0 characters (found 1). [LineLength]
Audit done.
Checkstyle ends with 4 errors.
````

Violations with BOM removed:
````
Starting audit...
[ERROR] M:\JavaBOM.java:1: Line is longer than 0 characters (found 24). [LineLength]
[ERROR] M:\JavaBOM.java:2: Line is longer than 0 characters (found 19). [LineLength]
[ERROR] M:\JavaBOM.java:3: Line is longer than 0 characters (found 5). [LineLength]
[ERROR] M:\JavaBOM.java:4: Line is longer than 0 characters (found 1). [LineLength]
Audit done.
Checkstyle ends with 4 errors.
````

First line's count increase by 1 character, though the physical text of the file is unchanged.

Reply:
Thanks for example. This situation could definitely be confusing for someone who is not familiar with BOM.